headerfs/store.go

```diff
@@ -439,6 +439,12 @@ func (h *BlockHeaderStore) CheckConnectivity() error {
 // ChainTip returns the best known block header and height for the
 // BlockHeaderStore.
 func (h *BlockHeaderStore) ChainTip() (*wire.BlockHeader, uint32, error) {
+	// Ensure index and flat file read are part of same tx.
+	tx, err := h.db.BeginReadTx()
+	defer tx.Rollback()
+	h.RLock()
+	defer h.RUnlock()
```

Comment:
Won't this cause deadlock? As `chainTip` itself also creates a new database transaction. Also the error there is never checked, I think this would mean that it doesn't compile?

Reply:
This compiles and hasn't deadlocked in tests on me yet. `chainTip` creates a read-only tx, which doesn't interfere with the tx created in this method.

Reply:
You're right, though, I need to check this error. Will fix now.